client/engine/messageservice/simple-tcp/simpletcp-messageservice.go

```diff
@@ -0,0 +1,90 @@
+package simpletcp
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+const (
+	connectionType = "tcp"
+)
+
+type SimpleTCPMessageService struct {
+	in  chan protocols.Message // for receiving messages from engine
+	out chan protocols.Message // for sending message to engine
+
+	myUrl string
+	peers map[types.Address]string
+}
+
+func NewSimpleTCPMessageService(myAddress types.Address, peers map[types.Address]string) *SimpleTCPMessageService {
+	h := &SimpleTCPMessageService{
+		in:    make(chan protocols.Message, 5),
+		out:   make(chan protocols.Message, 5),
+		peers: peers,
+		myUrl: peers[myAddress],
+	}
+	listener, err := net.Listen(connectionType, h.myUrl)
+	if err != nil {
+		panic(err)
+	}
+	go h.listenForIncoming(listener)
+	go h.listenForOutgoing()
+	return h
+
+}
+
+func (s *SimpleTCPMessageService) listenForOutgoing() {
+	for m := range s.in {
+		peer, ok := s.peers[m.To]
+
+		if !ok {
+			panic(fmt.Errorf("no peer port for %s", m.To))
+		}
+		raw, err := m.Serialize()
+		if err != nil {
+			panic(err)
+		}
+		conn, err := net.Dial(connectionType, peer)
```

Comment:
Here you've got a new connection per outgoing message. 

It is interesting to think about having a persistent TCP connection between peers, that could be reused to send messages. (I'm not suggesting we do this now).

In that model, I believe messages would be delivered to peers in the same order they were sent. This is relevant to this RFC https://www.notion.so/statechannels/RFC-14-Lean-on-message-delivery-guarantees-to-simplify-engine-protocols-81e11b814cfa46babae957dcdfdceae5.

Reply:
Interesting! That would guarantee that messages from a specific peer arrive in order. We would still have to deal with messages from different peers arriving in different orders (IE: We could get 3 messages from Bob before we get the first from Irene) but that might be fine.

It would introduce a little book keeping as a message service would have to negotiate a port for each peer instead of using a fixed port, but that probably shouldn't be too hard either.